rmc/utils/mpc.py

```diff
@@ -564,77 +698,127 @@ def calculate_fitted_scores(
             f"{intercept_str} not in model parameters! Please double check and rerun."
         )
 
-    logger.info("Annotating HT with MPC variables...")
-    variables = mpc_rel_vars.keys()
-    annotations = []
-    if "misbad" in variables:
-        logger.info("Getting missense badness annotation...")
-        mb_ht = misbad.versions[freeze].ht()
-        ht = ht.annotate(misbad=mb_ht[ht.ref, ht.alt].misbad)
-        annotations.append("misbad")
+    variable_dict = {k: v for k, v in mpc_rel_vars.items() if not interaction_char in k}
+    variables = variable_dict.keys()
+    interactions_dict = {k: v for k, v in mpc_rel_vars.items() if interaction_char in k}
 
-    if "oe" in variables:
-        annotations.append("oe")
+    # Select fields from context HT containing annotations to extract
+    # NOTE: `ref` and `alt` fields are also extracted for annotation of missense badness
+    context_ht_annots = {"oe", "polyphen"}
+    annots = context_ht_annots.intersection(variables)
+    logger.info("Annotating HT with MPC variables from context HT (%s)...", annots)
```

Comment:
It's possible it may be empty if the MPC model doesn't include any annotations we want to pull from the context table. But also check out the new code and lmk what you think!